### PR TITLE
Retire sacredtradition.am for feast names: lectionary engine

### PR DESCRIPTION
**Pull request by andylitalo**
## Summary

Feast-day names were the **last** runtime sacredtradition.am scrape in bahk. This retires it by computing feast names offline from the `armenian_lectionary` engine, mirroring the readings retirement in #461. With engine **1.2.3** it also serves the Armenian (`hy`) feast name offline (1.2.2 fixed feast/book name typos; 1.2.3 adds further reading fixes), so `name_hy` is populated too — closing the one gap the initial (English-only) retirement left, and removing the last reason feast names needed the live scrape.

Stacked on **#461** (`feat/retire-sacredtradition-readings`) — this PR is based on that branch so the diff shows only the feast changes. Merge/rebase after #461.

## Changes

- **`hub/services/feast_service.py`** — `get_feast_for_date(date_obj, church)` reads the engine's `"Liturgical Day"` in both `en` and `hy` and returns the same `{name, name_en, name_hy}` shape `scrape_feast` produced (or `None`). `name_hy` is the engine's Armenian name, left `None` only when the engine has no Armenian form (those keep falling back to English via `name_i18n` and can upgrade later). Filters engine placeholders, unsupported churches, and out-of-range dates (2001–2027).
- **`hub/utils.py`** — removes `scrape_feast`, `_fetch_sacredtradition`, `_stable_url_key`, the circuit-breaker constants, and the now-dead `urllib`/`hashlib`/`sentry_sdk` imports. `get_or_create_feast_for_date` lazily imports the service (avoids a circular import); it already backfills a *missing* `name_hy`, so existing feasts missing Armenian now self-heal.
- **`import_feasts`** management command points at the new service.
- **`requirements.txt`** — `armenian-lectionary >=1.2.3` (feast-name accuracy contract + offline `hy` feast/book names + typo/reading fixes).
- **sacredtradition.am scrub** — updated the now-inaccurate references on active surfaces: `Reading.text_hy_fetched_at` help text (**migration 0056**, chained after the base's `0055` drift-sync; help-text only, no schema change), the admin fieldset label, and the `fetch_armenian_reading_texts` command help/docstring (all compose from the offline `BibleVerse` corpus). Provenance/attribution notes on the corpus and the one-time `scripts/scrape_bible_hy.py` are intentionally retained.
- **Tests** — `test_feast_service.py` updated for the two-language engine call (asserts Armenian populates, and stays `None` when untranslated).

## Scope note: Armenian verse *text*

The engine provides Armenian **names** (feasts + reading book heads), not verse **text**. Armenian reading text still comes from the offline `BibleVerse` corpus (#460/#461) — there is no engine source for it, so the corpus stays. No runtime sacredtradition.am access remains anywhere.

## Verification

- 87 feast + lectionary/armenian-text tests pass (`manage.py test … --settings=tests.test_settings`); `manage.py makemigrations --check` clean (migration state fully in sync after rebasing onto the #461 base, which now carries the Day.church/FastIntention drift-sync).
- Real-engine check through the service (AL 1.2.3, not mocks): 2026-01-06 → `NATIVITY AND THEOPHANY…` / `ՏՕՆ ԾՆՆԴԵԱՆ ԵՒ ԱՍՏՈՒԱԾԱՅԱՅՏՆՈՒԹԵԱՆ…`, 2026-04-05 → `RESURRECTION… (Easter Sunday)` / `ՅԱՐՈՒԹԻՒՆ… (Զատիկ)`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: hub/services/feast_service.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    if not (LECTIONARY_MIN_YEAR <= date_obj.year <= LECTIONARY_MAX_YEAR):
+        logger.warning(
+            "Date %s is outside the validated lectionary range %d-%d; no feast returned.",
+            date_obj, LECTIONARY_MIN_YEAR, LECTIONARY_MAX_YEAR,
+        )
+        return None
+
+    result_en = armenian_lectionary.compute_armenian_lectionary(date_obj, language="en")
+    name_en = (result_en.get("Liturgical Day") or "").strip()
```
Comment (mattash) [inline hub/services/feast_service.py:45]: **[P1] Guard engine names against the `Feast.name` storage limit.** `Feast.name` is `CharField(max_length=256)`, but the exact 1.2.3 wheel returns 54 over-limit values across 2001–2027. For example, 2026-10-24 is 289 characters and 2026-12-03 is 257. These flow through `get_or_create_feast_for_date()` into `Feast.objects.get_or_create(... defaults={"name": name_en})`; PostgreSQL will raise `DataError`, making the API degrade to no feast and causing a range import to abort. SQLite tests do not enforce this. Please make the engine/storage contract safe and add a persistence regression test. Any schema expansion must follow the repo's migration hold lane with explicit Der Hayr approval and a separate rollback/data-preservation review.

File: hub/services/feast_service.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    if not name_en or any(marker in name_en for marker in _NON_FEAST_MARKERS):
```
Comment (mattash) [inline hub/services/feast_service.py]: **[P2] Do not turn this engine placeholder into a missing feast without recovering the real name.** The exact 1.2.3 wheel emits `"(movable ordinary-time reading)"` on five dates inside the supported range: 2011-02-04, 2011-02-06, 2011-02-09, 2011-02-11, and 2022-02-04. This branch therefore returns `None` for them, whereas the retired source returns actual data (for example, 2011-02-04 is `Fast day`). Please map/recover these values or fix the dependency before retiring the fallback, with a real-engine test for at least one affected date.

File: hub/services/feast_service.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        date_obj = date_obj.date()
+
+    if not (LECTIONARY_MIN_YEAR <= date_obj.year <= LECTIONARY_MAX_YEAR):
+        logger.warning(
+            "Date %s is outside the validated lectionary range %d-%d; no feast returned.",
+            date_obj, LECTIONARY_MIN_YEAR, LECTIONARY_MAX_YEAR,
+        )
+        return None
+
+    result_en = armenian_lectionary.compute_armenian_lectionary(date_obj, language="en")
```
Comment (mattash) [inline hub/services/feast_service.py:44]: **[P2] The pinned engine is not a drop-in source of feast names on 2011-02-13.** For that supported date, the exact 1.2.3 wheel returns `Eve of the Presentation of the Lord` in both `en` and `hy`, while the authoritative source returns `Fifth Sunday after Nativity — Eve of Fast of Catechumens` and has an Armenian form. This code consequently persists the wrong English feast and leaves `name_hy` empty. Please fix/upgrade the engine data or handle this exception, and add a real-engine regression test for the date.

**Review by mattash — CHANGES_REQUESTED** [review]
Requesting changes for three feast-data correctness issues found by exercising the exact `armenian-lectionary==1.2.3` wheel across the supported 2001–2027 range. The added mocked service tests do not cover these real-engine edge cases. Please address the persistence contract and add real-engine regression coverage for the affected dates. Note that any schema change is in the repo's migration hold lane and requires explicit Der Hayr approval plus separate rollback/data-preservation review.

File: hub/services/feast_service.py (hunk 3 above)
Comment (andylitalo) [reply hub/services/feast_service.py:44]: Upgrade to version 1.3.0 fixes this error.

File: hub/services/feast_service.py (hunk 2 above)
Comment (andylitalo) [reply hub/services/feast_service.py]: Upgrading to version 1.3.0 fixes this error.

File: hub/services/feast_service.py (hunk 1 above)
Comment (andylitalo) [reply hub/services/feast_service.py:45]: Fixed by extending `max_length` to 512 in [this merged PR](https://github.com/surp-hovhannes/bahk/pull/471).

Comment (andylitalo) [reply hub/services/feast_service.py:45]: Verified with armenian-lectionary 1.3.0 installed (`pip show armenian-lectionary` -> 1.3.0, pinned via `requirements.txt`). Re-ran the same over-256-char scan across the full engine-supported range (2001-2027): 54 dates still exceed 256 chars, but the max is unchanged at 289 ("The Twelve Holy Doctors of Church...", e.g. 2001-10-27) and 257 ("The Holy Fathers of Egypt...", e.g. 2001-12-06) -- none exceed 512. `Feast.name` is now `max_length=512` (hub/models.py:780, migration `0060_alter_feast_name.py`, landed in #471), with a comment recording exactly these two cases. `hub.tests.test_feast_service` passes (7/7).

File: hub/services/feast_service.py (hunk 2 above)
Comment (andylitalo) [reply hub/services/feast_service.py]: Verified with armenian-lectionary 1.3.0. Re-ran `compute_armenian_lectionary(date, language="en")` directly for all five flagged dates:

- 2011-02-04 -> `Fast day`
- 2011-02-06 -> `Fourth Sunday after Nativity`
- 2011-02-09 -> `Fast day`
- 2011-02-11 -> `Fast day`
- 2022-02-04 -> `Fast day`

None of these emit the `"(movable ordinary-time reading)"` placeholder anymore, so `get_feast_for_date` no longer returns `None` for them -- they resolve to real names as the retired scrape did.

File: hub/services/feast_service.py (hunk 3 above)
Comment (andylitalo) [reply hub/services/feast_service.py:44]: Verified with armenian-lectionary 1.3.0. Re-ran `compute_armenian_lectionary(date(2011,2,13), ...)` for both languages:

- en -> `Fifth Sunday after Nativity — Eve of Fast of Catechumens`
- hy -> `Ե կիւրակէ զկնի Ս. Ծննդեան — Բարեկենդան Առաջաւորաց պահոց`

This now matches the authoritative source's English name exactly, and an Armenian form is present (previously it fell back to English with `name_hy = None`). The `"Eve of the Presentation of the Lord"` value cited in the original review is no longer returned for this date.

**Review by mattash — APPROVED** [review]
love it
